shuup/xtheme/models.py

```diff
@@ -160,3 +173,22 @@ def update_settings(self, update_values):
 
     def __str__(self):
         return _("Theme configuration for %s") % self.theme_identifier
+
+
+class Snippet(models.Model):
+    """
+    Inject snippet code globally filtering by themes if configured
+    """
+    shop = models.ForeignKey("shuup.Shop", related_name="snippets")
+    location = models.CharField(max_length=64, verbose_name=_("location"))
+    snippet_type = models.CharField(max_length=20, verbose_name=_("snippet type"), choices=SnippetTypeChoices)
+    snippet = models.TextField(verbose_name=_("snippet"))
+    # list of theme identifiers that will be have this sniipet injected, if None, it means all themes
+    themes = SeparatedValuesField(verbose_name=_("themes"), blank=True, null=True)
```

Comment:
Why we need themes here?

Reply:
you can enable injection just for some themes

Reply:
Make sure optional then and add help text too.

Reply:
it is optional, I just added the help text to the form, I will move it to here